src/index.js

```diff
@@ -1,5 +1,16 @@
+import { AppContainer } from 'react-hot-loader';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const rootEl = document.getElementById('root');
+const render = Component =>
+  ReactDOM.render(
+    <AppContainer>
+      <Component />
+    </AppContainer>,
+    rootEl
+  );
+
+render(App);
+if (module.hot) module.hot.accept('./App', () => render(App));
```

Comment:
It should not, unless you are incorrectly transforming modules before webpack sees them.

You only need to `require` or `import()` again if you `require`d or `import()`ed the first time. Static imports are live and are always up-to-date (internally, webpack will re-require it before calling your callback).